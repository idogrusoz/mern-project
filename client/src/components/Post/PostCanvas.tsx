import { Grid, Paper, Typography, useMediaQuery, makeStyles } from "@material-ui/core";
import React, { FunctionComponent } from "react";

type PostCanvasType = {
    textContent: string;
    fontFamily: string;
    color: string;
    backgroundColor: string;
    fontWeight: number;
    fontSize: number;
    textAlign: "left" | "right" | "justify" | "center";
};

const PostCanvas: FunctionComponent<PostCanvasType> = ({
    backgroundColor,
    color,
    fontFamily,
    fontSize,
    fontWeight,
    textAlign,
    textContent,
}) => {
    const isBigScreen = useMediaQuery("(min-width: 650px)");
    const useStyles = makeStyles({
        wrapper: {
            display: "flex",
            justifyContent: "center",
            alignItems: "center",
            overflow: "hidden",
            width: !isBigScreen ? 330 : 600,
            height: !isBigScreen ? 220 : 400,
            backgroundColor,
            margin: "auto",
        },
        text: {
            display: "flex",
            justifyContent: "center",
            alignItems: "center",
            overflow: "hidden",
            width: !isBigScreen ? 330 : 600,
            height: !isBigScreen ? 220 : 400,
            fontFamily,
            color,
            fontWeight,
            fontSize: isBigScreen ? fontSize : fontSize * 0.55,
            margin: "auto",
            textAlign,
        },
    });
    const classes = useStyles();

    return (
        <Grid container justify="center">
            <Grid item md={12}>
                <Paper elevation={3} className={classes.wrapper}>
                    <Typography className={classes.text}>{textContent}</Typography>
                </Paper>
            </Grid>
        </Grid>
    );
};

export default PostCanvas;
