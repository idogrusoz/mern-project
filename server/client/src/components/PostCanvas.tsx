import { Grid, Paper, Typography, useMediaQuery } from "@material-ui/core";
import React, { FunctionComponent } from "react";

type PostCanvasType = {
    textContent: string;
    fontFamily: string;
    color: string;
    backgroundColor: string;
    fontWeight: number;
    fontSize: number;
    textAlign: "left" | "right" | "justify" | "center";
};

const PostCanvas: FunctionComponent<PostCanvasType> = ({
    backgroundColor,
    color,
    fontFamily,
    fontSize,
    fontWeight,
    textAlign,
    textContent,
}) => {
    const isBigScreen = useMediaQuery("(min-width: 650px)");

    return (
        <Grid container justify="center">
            <Grid item md={12} justify="center">
                <Paper
                    elevation={3}
                    style={{
                        display: "flex",
                        justifyContent: "center",
                        alignItems: "center",
                        overflow: "hidden",
                        width: !isBigScreen ? 330 : 600,
                        height: !isBigScreen ? 220 : 400,
                        backgroundColor,
                        margin: "auto",
                    }}
                >
                    <Typography
                        style={{
                            display: "flex",
                            justifyContent: "center",
                            alignItems: "center",
                            overflow: "hidden",
                            width: !isBigScreen ? 330 : 600,
                            height: !isBigScreen ? 220 : 400,
                            fontFamily,
                            color,
                            fontWeight,
                            fontSize,
                            margin: "auto",
                            textAlign,
                        }}
                    >
                        {textContent}
                    </Typography>
                </Paper>
            </Grid>
        </Grid>
    );
};

export default PostCanvas;
