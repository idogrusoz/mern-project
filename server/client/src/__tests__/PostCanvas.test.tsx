import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import PostCanvas from "../components/PostCanvas";

describe("PostCanvas test", () => {
    const props = {
        textContent: "",
        fontFamily: "Roboto",
        color: "#ffffff",
        backgroundColor: "#ffffff",
        fontWeight: 400,
        fontSize: 40,
        textAlign: "left" as "left" | "right" | "justify" | "center",
    };

    const wrapper = shallow(<PostCanvas {...props} />);
    it("renders without crash", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
