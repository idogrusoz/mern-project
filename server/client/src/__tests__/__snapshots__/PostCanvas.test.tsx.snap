// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCanvas test renders without crash 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={12}
  >
    <WithStyles(ForwardRef(Paper))
      className="makeStyles-wrapper-1"
      elevation={3}
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-text-2"
      />
    </WithStyles(ForwardRef(Paper))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
